checkstyle/validator/CopyrightValidator.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package validator;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.LineNumberReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Java class to validate copyright header presence in all source and test files.
+ */
+public class CopyrightValidator {
+
+    private static final String expectedAsyncProfilerCopyrightHeader = "/*\n" +
+                                " * Copyright The async-profiler authors\n" +
+                                " * SPDX-License-Identifier: Apache-2.0\n" +
+                                " */\n";
+
+    private static final String expectedJattachCopyrightHeader = "/*\n" +
+                                " * Copyright The jattach authors\n" +
+                                " * SPDX-License-Identifier: Apache-2.0\n" +
+                                " */\n";
+
+    private static final Set<String> fileExtensionsToCheck = new HashSet<>(Arrays.asList("c", "h", "cpp", "java"));
```

Comment:
It might be a good idea to move "which files to validate" concern to the `Makefile` and pass a list of files as args to `CopyrightValidator`. 

This way `CopyrightValidator` will only be responsible for validating the passed file(s), not navigating directories or filtering by extensions. On the other hand, we could include/exclude files to validate from the `Makefile` if need be.